virtcontainers/api.go

```diff
@@ -109,13 +117,21 @@ func createSandboxFromConfig(ctx context.Context, sandboxConfig SandboxConfig, f
 	defer func() {
 		if err != nil {
 			s.agent.stopSandbox(s)
+			s.agent.cleanupSandbox(s)
 		}
 	}()
 
-	if err := s.getAndStoreGuestDetails(); err != nil {
+	if err = s.getAndStoreGuestDetails(); err != nil {
 		return nil, err
 	}
 
+	// rollback all sandbox related local filesystem storages
+	defer func() {
+		if err != nil {
+			s.storage.deleteSandboxResources(s.id, nil)
```

Comment:
Why not call `s.Delete()` ? It should consist everything to make a sandbox clean, if not, we should add more cleanup logics there.

Reply:
@sboeuf I call `s.storage.deleteSandboxResources()` here because in `s.getAndStoreGuestDetails()` it calls `s.storage.storeSandboxResource(s.id, stateFileType, s.state)`